common/short-hex-str/src/lib.rs

```diff
@@ -1,5 +1,6 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
+#![allow(clippy::integer_arithmetic)]
```

Comment:
It's better if we allow individual functions or block of code (if that's possible) instead of the whole file here